### PR TITLE
Feature/add minimal k8s scheduler

### PR DESCRIPTION
**Pull request by Mtze**
Steps to test 

- Start Minikube 
```bash
minikube start
```
- Start API and Queue 
```
docker compose -f docker-compose.yml -f docker-compose.dev.yml up redi
s fluentbit hadesAPI
```
- Start K8s scheduler 
```
DEBUG=true HADES_EXECUTOR=k8s go run ./HadesScheduler
```
- Schedule a successful job with bruno 
- Schedule a failing job with bruno 
- Observe running pods: 
```
k -n hades-executor get po
# you may look at the container logs - the "-c 2" indicates that you will look at the second init container which is the build execution container in the bruno example
k -n hades-executor logs -f <pod name> -c 2
```

### CODE REVIEW

File: HadesScheduler/k8s/job.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+// Schedules a Hades Job on the Kubernetes cluster
+func (k8sJob K8sJob) execute(ctx context.Context) error {
+	slog.Info("Scheduling job", "id", k8sJob.ID)
+
+	slog.Debug("Create buildscript ConfigMap")
+	configMap := k8sJob.configMapSpec()
+	slog.Debug("ConfigMap spec", "config_map", configMap)
+
+	slog.Debug("Apply buildscirpt ConfigMap to Kubernetes")
```
Comment (robertjndw) [inline HadesScheduler/k8s/job.go:30]:
```suggestion
	slog.Debug("Apply buildscript ConfigMap to Kubernetes")
```

File: HadesScheduler/k8s/job.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	cm, err := k8sJob.k8sClient.CoreV1().ConfigMaps(k8sJob.namespace).Create(ctx, configMap, metav1.CreateOptions{})
+	if err != nil {
+		slog.With("error", err).Error("Failed to create ConfigMap")
+		return err
+	}
+	slog.Info("Successfully created buildscirpt ConfigMap", "name", cm.Name)
```
Comment (robertjndw) [inline HadesScheduler/k8s/job.go:36]:
```suggestion
	slog.Debug("Successfully created buildscript ConfigMap", "name", cm.Name)
```

File: HadesScheduler/k8s/job.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+// Creates a ConfigMapSpec containing the build script of each step of the job
+// Based on this configMap, individual step volumes are created and mounted inside the respective container
+func (k8sJob K8sJob) configMapSpec() *corev1.ConfigMap {
+	configMap := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      k8sJob.ID.String(),
+			Namespace: k8sJob.namespace,
+		},
+		Data: map[string]string{},
```
Comment (robertjndw) [inline HadesScheduler/k8s/job.go:79]: Do we need to specify it? Or is the default always am empty map?

File: HadesScheduler/k8s/k8s.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	// Load the user provided Kubernetes configuration
+	var k8sCfg K8sConfig
+	utils.LoadConfig(&k8sCfg)
+	slog.Debug("Kubernetes config", "config", k8sCfg)
+
+	// Initialize the Kubernetes scheduler
+	slog.Info("Initializing Kubernetes client")
+	scheduler := initializeClusterAccess(k8sCfg)
+
+	// TODO: Check cluster connection and print cluster nodes to log
```
Comment (robertjndw) [inline HadesScheduler/k8s/k8s.go:51]: Ping

File: HadesScheduler/k8s/namespace.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	"log/slog"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func createNamespace(ctx context.Context, clientset *kubernetes.Clientset, namespace string) (*corev1.Namespace, error) {
+	// Create a namespace in the Kubernetes cluster
+	slog.Info("Creating namespace", "namespace", namespace)
```
Comment (robertjndw) [inline HadesScheduler/k8s/namespace.go:15]: I feel like we're spaming the console too much if we always have a `creating ...` and `created ...` log message at the info channel. Personally, I prefer the `created ...` log msg more and would put only this in the Info output
```suggestion
	slog.Debug("Creating namespace", "namespace", namespace)
```

File: HadesScheduler/k8s/namespace.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+		return nil, err
+	}
+	slog.Info("Namespace created", "namespace", namespace)
+
+	return ns, nil
+}
+
+func deleteNamespace(ctx context.Context, clientset *kubernetes.Clientset, namespace string) error {
+	// Delete a namespace in the Kubernetes cluster
+	slog.Info("Deleting namespace", "namespace", namespace)
```
Comment (robertjndw) [inline HadesScheduler/k8s/namespace.go:36]:
```suggestion
	slog.Debug("Deleting namespace", "namespace", namespace)
```

File: HadesScheduler/k8s/k8s.go (hunk 4 above)
Comment (Mtze) [reply HadesScheduler/k8s/k8s.go:51]: Will be done in a follow up. This is just convenient for the admin :P

File: HadesScheduler/k8s/namespace.go (hunk 5 above)
Comment (Mtze) [reply HadesScheduler/k8s/namespace.go]: I agree

File: HadesScheduler/k8s/job.go (hunk 3 above)
Comment (Mtze) [reply HadesScheduler/k8s/job.go:79]: Yes - We put in the steps later which require the map to be initialized

**Review by robertjndw — APPROVED** [review]

**Pull request merged**